lib/e3/ecl/fortio.h

```diff
@@ -166,6 +166,70 @@ int eclfio_get( FILE*, const char* opts, int32_t* recordsize, void* record );
  */
 int eclfio_put( FILE*, const char* opts, int nmemb, const void* );
 
+/*
+ * By default, eclipse writes arrays in blocks of N elements, that is, a 5000
+ * element integer array is written in e.g. 5 blocks of 1000 elements each, but
+ * considered the same logical vector spanning multiple physical fortran
+ * blocks.
+ *
+ * The eclfio_get/put functions has no such behavorial restriction, and
+ * happily writes longer arrays in a single physical block.
+ *
+ * To automate handling arrays spanning multiple physical blocks, use
+ * eclfio_array_get/put. The ecl_default_blocksize function returns a blocksize
+ * compatible with eclipse's default sizes.
+ *
+ * ecl_default_blocksize expects one of the values in ecl_default_blocksizes
+ * enum, otherwise a negative value is returned.
+ */
+#define ECL_DEFAULT_BLOCKSIZE_NUMERIC 1000
+#define ECL_DEFAULT_BLOCKSIZE_STRING  105
+int ecl_default_blocksize( int );
+
+/*
+ * eclfio_array_get/put behave differently w.r.t. errors than their simpler
+ * relatives eclfio_get/put.
+ *
+ * When an operation cannot be successfully completed, eclfio_array does *not*
```

Comment:
I think these functions will be the main atomic units; so personally I think argument parsing, roll-back guarantees and so on should be at this level. The fortio level is not interesting for the user,

Reply:
> I think these functions will be the main atomic units; so personally I think argument parsing, roll-back guarantees and so on should be at this level. The fortio level is not interesting for the user,

Who's _the user_?

I have mixed feelings about this. If rollback is done on array, then we'd either drop or have to communicate _which_ block failed, but if it only rolls back to the point that failed, then there's immediate recoverability.

Also, that means reading record headers can read with `eclfio_get` directly, with rollback semantics, which is nice.

That being said, `array_get` *will* roll back if nmemb < size.

Reply:
> Who's the user?

The "user" in this case is just code - probably same library, slighter higher up in the stack.

> If rollback is done on array, then we'd either drop or have to communicate which block failed, but if it only rolls back to the point that failed, then there's immediate recoverability.

Yes - my point is that array is the relevant atomic unit - in all(?) contexts; i.e. I do not see it as interesting functionality to roll back to middle of an array - i.e. a record.

> Also, that means reading record headers can read with eclfio_get directly, with rollback semantics, which is nice.

I agree that reading headers with rollback semantics is nice - and important, but they can - and should(?) - be implemented as array_get of 16 characters - i.e. again array smenatics; although the array in this case always only consists of one block.

Reply:
> The "user" in this case is just code - probably same library, slighter higher up in the stack.

Sure, which is why I personally like this. It allows to check that specific block with sizeof, and quickly determine that the fortran block was of the wrong size, and maybe even pad the output array accordingly for robustness.

> Yes - my point is that array is the relevant atomic unit - in all(?) contexts; i.e. I do not see it as interesting functionality to roll back to middle of an array - i.e. a record.

The main use case is error handling, and I think the cost in interface is worth it personally.